### ISSUE TITLE
Avoid merge commits if possible

### ISSUE THREAD

**Issue by purpleidea** [rgbkrk/libvirt-go#120]
Yesterday I submitted a very simple patch which applied directly to git master:

https://github.com/rgbkrk/libvirt-go/commit/dd29b1f4579842e77773ebfeb765c1718d0a4881

Thanks for merging it, however it was done with a merge commit:

https://github.com/rgbkrk/libvirt-go/commit/644521deb7a2b74ca4dfa5c928e007b3587a5110

I'd like to propose that we avoid merge commits as much as possible, and in particular when there is only one pending patch, we should get it rebase to git master so that it can apply with a fast forward.

IOW, don't click the green merge button in it's current state. Much better to change the button to default to rebase if we plan on using it at all, or apply these kinds of patches with the cli.

Just my $0.02

Cheers!

**Comment by purpleidea** [comment]
Here's a nice example for this: https://github.com/rgbkrk/libvirt-go/pull/121

**Comment by vincentbernat** [comment]
I don't have a strong opinion on that. Rebase or regular merge are fine by me. Squash is not.

**Comment by purpleidea** [comment]
Yes, I agree we shouldn't squash. Fast forwarding is my preferred approach when available. Keeps the history logical and easy to understand.

**Comment by rgbkrk** [comment]
Rebase is good, we can set that as the default on the repo.
